### ISSUE TITLE
Pass lint

### ISSUE THREAD

**Issue by breerly** [thriftrw/thriftrw-go#127]
Wondering how possible it is for generated code to pass lint.

The following thrift file produced generated code that had a slew of warnings:

```
service Echo {
    Pong echo(1: Ping ping) (
        ttlms = '100'
    )
}

struct Ping {
    1: required string beep
}

struct Pong {
    1: required string boop
}
```

```
crossdock/thrift/echo/service/echo/echo.go|33 col 6 warning| exported type EchoArgs should have comment or be unexported
crossdock/thrift/echo/service/echo/echo.go|33 col 6 warning| type name will be used as echo.EchoArgs by other packages, and that stutters; consider calling this Args
crossdock/thrift/echo/service/echo/echo.go|37 col 1 warning| exported method EchoArgs.ToWire should have comment or be unexported
crossdock/thrift/echo/service/echo/echo.go|47 col 6 warning| don't use underscores in Go names; func _Ping_Read should be _PingRead
crossdock/thrift/echo/service/echo/echo.go|53 col 1 warning| exported method EchoArgs.FromWire should have comment or be unexported
crossdock/thrift/echo/service/echo/echo.go|79 col 6 warning| exported type EchoResult should have comment or be unexported
crossdock/thrift/echo/service/echo/echo.go|79 col 6 warning| type name will be used as echo.EchoResult by other packages, and that stutters; consider calling this Result
crossdock/thrift/echo/service/echo/echo.go|83 col 1 warning| exported method EchoResult.ToWire should have comment or be unexported
crossdock/thrift/echo/service/echo/echo.go|93 col 6 warning| don't use underscores in Go names; func _Pong_Read should be _PongRead
crossdock/thrift/echo/service/echo/echo.go|99 col 1 warning| exported method EchoResult.FromWire should have comment or be unexported
crossdock/thrift/echo/service/echo/echo.go|125 col 5 warning| exported var EchoHelper should have comment or be unexported
```

Thoughts?

**Comment by blampe** [comment]
It's easy enough to add generated code to lint-ignore lists, so this is more like a nice-to-have.

**Comment by prashantv** [comment]
Agree that this is a nice-to-have. It's blocked on #100

**Comment by breerly** [comment]
Agreed.

In the same way that the generated code gets ran through tests, we should also run that code through the other go tools, eg lint, vet, race checker, etc...

**Comment by abhinav** [comment]
I think at least part of this is a 1.0 requirement. Being able to parse the docstrings in Thrift files and including them as comments in the generated code will help pass at least some linter errors.

**Comment by abhinav** [comment]
Code generated by ThriftRW now has comments. Most of it should pass lint. The
rest, especially the `_`-named types, are "wontfix" because we intentionally
chose names that wouldn't conflict with user-specified idiomatic type names.
